src/Repl/Decode.tsx

```diff
@@ -0,0 +1,199 @@
+  return "(" + tuple.map((e) => decodeValue(e)).join(",") + ")";
+}
+
+
+/*
+ * decodeBool
+ * Decodes a Bool
+ *
+ * decoded : Decoded response that contains a bool
+ */
+const decodeBool = (decoded) => {
+  return decoded ? "True" : "False";
+}
+
+
+/*
+ * decodeExprType
+ * Decodes the type of the given expression
+ *
+ * latest : JSON obj of the latest expression
+ */
+const decodeExprType = (typ) => {
+  // report this type, if marked with ':t ' from the cmd line
+  // get first letter in lowercase
+  let fl = typ[0].toLowerCase();
+  let res = " is";
+
+  if(fl === 'a' || fl === 'e' || fl === 'i' || fl === 'o' || fl === 'u') {
+    // vowel sub y, use 'an'
+    res += " an ";
+
+  } else {
+    // consonant
+    res += " a ";
+
+  }
+  return res + typ;
+}
+
+
+/*
+ * decodeError
+ * Decodes an error related to a BoGL API request
+ *
+ * error :      Obj of error that has occurred
+ * respStatus : Status code of the response
+ */
+const decodeError = (error, respStatus : number) => {
+
+  console.error(error)
+
+  if((error instanceof SyntaxError || (error.name && error.name === "SyntaxError")) && respStatus === 504) {
+    // gateway timeout
+    return " 🤖 BoGL Says: Unable to finish running your program, or not currently online. Double check your code, or check back later! ";
+
+  } else if((error instanceof SyntaxError || (error.name && error.name === "SyntaxError"))) {
+    // bad parse error
+    return " 🤖 BoGL Says: I couldn't understand your program. Please double check it and try again! ";
+
+  } else if((error instanceof TypeError || (error.name && error.name === "TypeError")) && respStatus === 0) {
+    // likely JS disabled
+    return " 🤖 BoGL Says: Unable to execute your program. Make sure that Javascript is enabled and try again! ";
+
+  } else if((error instanceof TypeError || (error.name && error.name === "TypeError"))) {
+    // something else?
+    return " 🤖 BoGL Says: Unable to execute your program, please double check your code and try again. ";
+
+  } else {
+    // general error
+    return " 🤖 BoGL Says: An error occurred: " + error + " ";
+
+  }
+}
+
+
+/*
+ * decodeReponse
+ * Decodes a response from the BoGL API, returned a decoded object
+ *
+ * reponses: JSON obj containing the responses returned
+ */
+const decodeResponse = (responses: any) => {
+  // get latest response
+  const latestResponse = responses[responses.length - 1]
+
+  const category = latestResponse["tag"]
+  const contents = latestResponse["contents"]
+  let typ = ""
+  if(contents[1] !== undefined && contents[1]["type"] !== undefined) {
+    typ = contents[1]["type"]
+  }
+
+  let value = ""
+  if(contents[1] !== undefined && contents[1]["value"] !== undefined) {
+    value = contents[1]["value"]
+  }
+
+  let boards   = ""
+  if(category === "SpielPrompt") {
+    if(contents[0] !== undefined && contents[0]["value"] !== undefined) {
+      boards = contents.reduce((tot,b) => tot + decodeBoard(b["value"]) + "\n")
+    }
+
+  } else {
```

Comment:
This comment is what led me to find the decoding bug. All boards that are on the display buffer should be printed. The back end filters predecessors, it does not need to be done here.

Reply:
The board logic is quite old apparently. I was able to tear out most of it and replace it with what I've done so far. It was parsing things redundantly in some cases too. That's good to know they are filtered on the backend, I was not aware of that, should simplify things.

All boards that are the result of in-between places and such come back in response in this `boards` value. They just need to collected and parsed like all other values, and reported in the Repl accordingly.